lib/scan.ts

```diff
@@ -1,44 +1,76 @@
 import * as fs from 'fs';
-import { fromUrl } from 'hosted-git-info';
 import * as path from 'path';
+
+import { Facts, Options, PluginResponse, ScanResult } from './types';
+import { SignatureResult } from './types';
+import { debug } from './debug';
 import { find } from './find';
+import { fromUrl } from 'hosted-git-info';
+import { getSignature } from './signatures';
 import { getTarget } from './git';
-import { hash } from './hash';
-import { debug } from './debug';
-import {
-  ScanResult,
-  Options,
-  Fingerprint,
-  Facts,
-  PluginResponse,
-} from './types';
+
+const processSignatures = async (
+  signatures: Promise<SignatureResult | null>[],
+  allSignatures: SignatureResult[],
+) => {
+  const signaturesResult = await Promise.all(signatures);
```

Comment:
Do you want unbounded concurrency here? This will cause node to process every single signatureResult as fast as possible at the same time

Reply:
I'll let @anthogez address this, but I just want to share the observation that the amount of concurrency is controlled by `signatureConcurrency`. Antonio told me that he did some testing to find an optimal value for this.

Reply:
I don't understand though; this is unbounded. You're calling Promise.all... why not use p-map like we do in the rest of Snyk? 😄

Reply:
@robcresswell, I see what you mean now
`concurrency` limitation with e.g. const result = await pMap(sites, mapper, {concurrency: 20});

instead of use `promise.all` thanks!

Reply:
I've refactored now to use pMap. @robcresswell or @anthogez, it would be great if you could review the changes, as I've never used pMap before. Here is the specific commit with the changes: https://github.com/snyk/snyk-cpp-plugin/pull/34/commits/380ecc9d9ca77f8d34b04e2edeb40ea170887d2c